server/methods/catalog.js

```diff
@@ -675,17 +677,39 @@ Meteor.methods({
     // out if nothing to delete
     if (!Array.isArray(toDelete) || toDelete.length === 0) return false;
 
-    // Flag the variant and all its children as deleted in Revisions collection.
-    toDelete.forEach((product) => {
-      Hooks.Events.run("beforeRemoveCatalogProduct", product, { userId: this.userId });
-      Hooks.Events.run("afterRemoveCatalogProduct", this.userId, product);
-    });
+    if (RevisionApi.isRevisionControlEnabled()) {
+      // Flag the variant and all its children as deleted in Revisions collection.
+      // eslint-disable-next-line no-shadow
+      toDelete.forEach((variant) => {
+        Hooks.Events.run("beforeRemoveCatalogProduct", variant, { userId: this.userId });
+        Hooks.Events.run("afterRemoveCatalogProduct", this.userId, variant);
+      });
+    } else {
+      // eslint-disable-next-line no-shadow
```

Comment:
I wonder if there shouldn't be some hook here too in case someone wants to track deletions (e.g. for syncing with an external service)?

Reply:
Ok, right now the hooks are doing revisions related stuff, AFAIK. Probably it would be better to run the hooks unconditionally and put the switch in there.